cirq-google/cirq_google/devices/grid_device_test.py

```diff
@@ -465,6 +465,33 @@ def test_grid_device_validate_operations_negative():
         )
 
 
+def test_grid_device_validate_operation_coupler():
+    _, spec = _create_device_spec_with_horizontal_couplings()
+    device = cirq_google.GridDevice.from_proto(spec)
+
+    g = cirq_google.InternalGate(
+        gate_name="DetuneCoupler", gate_module='internal_module', num_qubits=1, freq=5.5
+    )
+    for y in range(GRID_HEIGHT):
+        # Valid couplers
+        coupler = Coupler(cirq.GridQubit(y, 0), cirq.GridQubit(y, 1))
+        device.validate_operation(g(coupler))
+        coupler = Coupler(cirq.GridQubit(y, 1), cirq.GridQubit(y, 0))
+        device.validate_operation(g(coupler))
+        # One coupler off grid
+        coupler = Coupler(cirq.GridQubit(y, 1), cirq.GridQubit(y, 2))
+        with pytest.raises(ValueError, match="Qubits on coupler not on device"):
+            device.validate_operation(g(coupler))
+        # Both couplers off grid
+        coupler = Coupler(cirq.GridQubit(y, 2), cirq.GridQubit(y, 3))
+        with pytest.raises(ValueError, match="Qubits on coupler not on device"):
+            device.validate_operation(g(coupler))
+        # Vertical Coupler
+        coupler = Coupler(cirq.GridQubit(y, 0), cirq.GridQubit((y + 1) % GRID_HEIGHT, 0))
```

Comment:
Why are vertical couplers not supported here? From the [slides](https://docs.google.com/presentation/d/1bbhuNsB4UzdrOEgbvQnc2CH1WBfY4boqstJ19TUXT0E/edit#slide=id.g2787ca194a7_0_18) this looks to be a valid use case.. can you elaborate?

Reply:
It is a test case.  The device for this test is generated with only horizontal couplers so that we can test pairs that are not on the device.  See line 469

Reply:
Thanks for clarifying!